test/units/testsuite-05.eff-limit.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -eux
+set -o pipefail
+
+# shellcheck source=test/units/util.sh
+. "$(dirname "$0")"/util.sh
+
+pre=test05
+cat >/run/systemd/system/"$pre"alpha.slice <<EOF
+[Slice]
+MemoryMax=40M
+MemoryHigh=40M
+TasksMax=400
+EOF
+
+cat >/run/systemd/system/"$pre"alpha-beta.slice <<EOF
+[Slice]
+MemoryMax=10M
+MemoryHigh=10M
+TasksMax=100
+EOF
```

Comment:
How about one more slice or reusing the same to test the memory high, which should effectively tell users "You are approaching the limit"

```suggestion
cat >/run/systemd/system/"$pre"alpha-beta.slice <<EOF
[Slice]
MemoryMax=10M
MemoryHigh=8M
TasksMax=100
EOF
```

While on that note, can `MemoryHigh` be a percentage?

Reply:
Not sure I get it -- do you mean that `MemoryHighEffective=` should be capped by `MemoryMaxEffective=`?

Re: percentages -- why would be percetages better? (IIUC, all values of given limit would have to be in percents so that the test compares apples to apples (as otherwise absolute values would depend on the environment the test runs in).

Reply:
> Not sure I get it -- do you mean that `MemoryHighEffective=` should be capped by `MemoryMaxEffective=`?

Yes, from what I remember of the story, one of the things users were complaining about was that they didn't have any facility that would tell them, "You're approaching the absolute limit," so by the time the process would hit the ceiling, it was already too late for them.

or am I expecting systemd to emit a signal in the wrong place?

> Re: percentages -- why would be percetages better? (IIUC, all values of given limit would have to be in percents so that the test compares apples to apples (as otherwise absolute values would depend on the environment the test runs in).

It is not necessarily better; it was an idea when looking at it

Reply:
The approaching notification sounds better suited with memory PSI ([see also](https://github.com/systemd/systemd/pull/28797#discussion_r1293834680) and [oomd RFE](https://github.com/systemd/systemd/issues/23606#issuecomment-1148906370)).

I'd rather keep this PR for simple effective limits.